config/everything.mk

```diff
@@ -184,9 +188,17 @@ run-$(3): run-$(1)
 
 endef
 
+ifeq "$(FD_HAS_MAIN)" "1"
 make-bin       = $(eval $(call _make-exe,$(1),$(2),$(3),bin))
 make-unit-test = $(eval $(call _make-exe,$(1),$(2),$(3),unit-test))
+make-fuzz-test =
 run-unit-test = $(eval $(call _run-unit-test,$(1),$(2),unit-test))
+else
+make-bin =
+make-unit-test =
+make-fuzz-test = $(eval $(call _make-exe,$(1),$(2),$(3),fuzz-test))
+run-unit-test =
+endif
```

Comment:
Instead of doing this conditional compilation and requiring this flag, it might be cleaner to just have the fuzzing script call `make fuzz-test`. Feels like it should be up to the caller to choose what should be built, not have `make` magically exclude stuff.